www/js/controllers.js

```diff
@@ -55,11 +55,29 @@ angular.module('starter.controllers', [])
     }
   }
 
-  $scope.addToRecentSearches = function() {
-    RecentSearches.push($scope.searchText);
-    Store.set('sikher_recent', RecentSearches);
-  }
-
+	$scope.addToRecentSearches = function () {
+		//remove from recent search if it was same as new one
+		for (var i = 0; i < RecentSearches.length; i++) {
```

Comment:
This way this code currently works means that it would:
1. remove the duplicate
2. add the duplicate back in later on with the `RecentSearches.push({ ... })`

So instead of this you may want to consider something like:

```
if (RecentSearches[i].searchText === $scope.searchText && RecentSearches[i].searchType === $scope.search) {
    return;
}
```